### DISCUSSION TITLE
Checkout without submodules

### DISCUSSION THREAD

**Discussion by w010** [TYPO3/Surf#779] [Q&A] [answered]
When I set _recursiveSubmodules = false_  in task options, it checks them out on server anyway. I need the submodules to be omitted, they are used only for development. These are such things like .ddev, frontend mockup, etc. I know they can be excluded later in _rsyncExcludes_, but it takes much time to checkout all these unnecessary subrepos, and there's complications with some private repo permissions. How to make Surf just omit them?
This should work, but it doesn't help:
```php
$deployment 
    ->addApplication($application) 
    ->onInitialize(function () use ($deployment, $application) {
        $deployment->getWorkflow()
            ->setTaskOptions(\TYPO3\Surf\Task\Package\GitTask::class, [
                'recursiveSubmodules' => false,
            ])
```
Also tried active = false in .gitmodules, but it's not respected.
How to make that work?

**Comment by simonschaufi** [comment, marked as answer]
The flag `recursiveSubmodules` only disables the submodules of the submodules, so the second level so to speak but NOT the first level. Disabling the checkout of the first level submodules is currently not supported. Would you sponsor this feature?
